### ISSUE TITLE
Preedit disappears and appears elsewhere when the cursor is moved with a mouse

### ISSUE THREAD

**Issue by mike-fabian** [OpenBangla/OpenBangla-Keyboard#298]
See:

https://github.com/ibus/ibus/issues/2420#issuecomment-1176413549

which I reproduce here for convenience:

> Otherwise, when I move the cursor by mouse from the place where I was writing last, if I move the cursor elsewhere; Then that word also disappears, then wherever I put the mouse cursor, that word goes.

That is also a bug in OpenBangla:

https://github.com/OpenBangla/OpenBangla-Keyboard/blob/master/src/engine/ibus/main.cpp#L46

uses:

```
ibus_engine_update_preedit_text(engine, text, ibus_text_get_length(text), TRUE);
```

It should instead use:

```
ibus_engine_update_preedit_text_with_mode(engine, text, ibus_text_get_length(text), TRUE, IBUS_ENGINE_PREEDIT_COMMIT)
```

If that is used and you move the cursor elsewhere, the word currently in preedit will not disappear but it will be committed instead. And therefore also not reappear elsewhere.

**Comment by mike-fabian** [comment]
See: https://github.com/ibus/ibus/blob/master/src/ibusengine.h#L228

**Comment by mominul** [comment]
Thanks a lot @mike-fabian ! This bug is now fixed and I have tested it locally. #300
This fix will be available in the 3.0.0 version!
